client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import { __experimentalCreateInterpolateElement } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '../../stores/onboard';
+import { Step, usePath } from '../../path';
+import Arrow from './arrow';
+
+interface Props {
+	inputRef: React.RefObject< HTMLInputElement >;
+}
+const SiteTitle: React.FunctionComponent< Props > = ( { inputRef } ) => {
+	const { __: NO__ } = useI18n();
+	const { siteTitle } = useSelect( select => select( STORE_KEY ).getState() );
+	const { setSiteTitle } = useDispatch( STORE_KEY );
+	const history = useHistory();
+	const makePath = usePath();
+
+	const handleChange = ( e: React.ChangeEvent< HTMLInputElement > ) =>
+		setSiteTitle( e.target.value.trim().length ? e.target.value : '' );
```

Comment:
I might be missing something, but isn't
`e.target.value.trim().length ? e.target.value : ''` ≡ `e.target.value.trim()`

Reply:
I think this and your next point could likely be cleaned up, but I believe some of the additional complexity is necessary.

From memory, we added some of this because we want to avoid erasing characters as users type. When they type ` `, the input shouldn't contain an empty string. At the same time, we want to avoid setting the site title to a string that only contains whitespace.

Reply:
Ha! This is a clever way to handle this. I think copying your comment to the code could help.